analyzers/src/SonarAnalyzer.CSharp/Rules/ForLoopConditionAlwaysFalse.cs

```diff
@@ -120,72 +120,64 @@ private static bool IsLogicalNot(ExpressionSyntax expression, out PrefixUnaryExp
+            if (ExpressionNumericConverter.TryGetConstantDoubleValue(expression, out parsedValue)
                 || (expression is SimpleNameSyntax simpleName
-                    && variableNameToIntegerValue.TryGetValue(simpleName.Identifier.ValueText, out intValue)))
+                    && variableNameToDoubleValue.TryGetValue(simpleName.Identifier.ValueText, out parsedValue)))
             {
                 return true;
             }
 
-            intValue = default(int);
+            parsedValue = default;
             return false;
         }
 
         /// <summary>
-        /// Retrieves the mapping of variable names to their integer value from the variable declaration part of a for loop.
+        /// Retrieves the mapping of variable names to their double value from the variable declaration part of a for loop.
         /// This will find the mapping for such cases:
         /// <code>
         /// for (var i = 0;;) {}
         /// </code>
         /// </summary>
-        private static IDictionary<string, int> GetVariableDeclarationMapping(VariableDeclarationSyntax variableDeclarationSyntax)
+        private static IDictionary<string, double> VariableDeclarationMapping(VariableDeclarationSyntax variableDeclarationSyntax)
         {
-            var loopInitializerValues = new Dictionary<string, int>();
+            var loopInitializerValues = new Dictionary<string, double>();
             if (variableDeclarationSyntax != null)
             {
                 foreach (var variableDeclaration in variableDeclarationSyntax.Variables)
                 {
                     if (variableDeclaration.Initializer is EqualsValueClauseSyntax initializer
-                        && ExpressionNumericConverter.TryGetConstantIntValue(initializer.Value, out var intValue))
+                        && ExpressionNumericConverter.TryGetConstantDoubleValue(initializer.Value, out var doubleValue))
                     {
-                        loopInitializerValues.Add(variableDeclaration.Identifier.ValueText, intValue);
+                        loopInitializerValues.Add(variableDeclaration.Identifier.ValueText, doubleValue);
                     }
                 }
             }
             return loopInitializerValues;
         }
 
         /// <summary>
-        /// Retrieves the mapping of variable names to their integer value from the initializer part of a for loop.
+        /// Retrieves the mapping of variable names to their double value from the initializer part of a for loop.
         /// This will find the mapping for such cases:
         /// <code>
         /// int i;
         /// for (i = 0;;) {}
         /// </code>
         /// </summary>
-        private static IDictionary<string, int> GetLoopInitializerMapping(IEnumerable<ExpressionSyntax> initializers)
+        private static IDictionary<string, double> LoopInitializerMapping(IEnumerable<ExpressionSyntax> initializers)
```

Comment:
I think we should use decimal instead of double to avoid potential problems caused by equality checks. See also the 2 bugs raised.